### PR TITLE
Respect the original terminal tab order after drag multiple tabs

### LINKED ISSUE #224070: Not able to move multi tab in the terminal list at the same time
- VS Code Version: 
Version: 1.91.1 (Universal)
Commit: f1e16e1e6214d7c44d078b1f0607b2388f29d729
Date: 2024-07-09T22:07:54.982Z
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin arm64 23.5.0
- OS Version: MacOS Sonoma 14.5

### Current behaviour
CleanShot 2024-07-27 at 15 08 12

### Expect behaviour
CleanShot 2024-07-27 at 19 47 14

### PR DESCRIPTION
**Pull request by BABA983**
https://github.com/microsoft/vscode/pull/224301#issuecomment-2263688251
Close #224070

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalGroup.ts [hunk 1]
```diff
@@ -409,22 +409,20 @@ export class TerminalGroup extends Disposable implements ITerminalGroup {
...
 	moveInstance(instances: ITerminalInstance | ITerminalInstance[], index: number, position: 'before' | 'after'): void {
-		const instances2 = Array.isArray(instances) ? instances : [instances];
-		for (const instance of instances2) {
-			const index = this.terminalInstances.indexOf(instance);
-			if (index === -1) {
-				return;
-			}
+		const instancesArray = Array.isArray(instances) ? instances : [instances];
+		const hasInvalidInstance = instancesArray.some(instance => !this.terminalInstances.includes(instance));
+		if (hasInvalidInstance) {
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalGroup.ts:414]: shouldn't this be `if (!hasValidInstance)`?

**Comment by meganrogge** [comment]
in oss based on your branch, still seeing the order can be wrong:

https://github.com/user-attachments/assets/7d340598-956d-49fb-88c8-ad359125717e

**Comment by BABA983** [comment]
> in oss based on your branch, still seeing the order can be wrong:
> 
>  group.mov

Just pushed a commit, should be good now

**Comment by meganrogge** [comment]
thanks, will test it again. can you reply to https://github.com/microsoft/vscode/pull/224591#discussion_r1702142439?

File: src/vs/workbench/contrib/terminal/browser/terminalGroup.ts (hunk 1 above)
Comment (BABA983) [reply src/vs/workbench/contrib/terminal/browser/terminalGroup.ts:414]: I think the `hasInvalidInstance ` was trying to indicate the source array has at least one instance not found in `this. terminalInstances`. But I guess `!hasValidInstance` and `hasInvalidInstance` are kind of the same?

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalGroup.ts:414]: oh, I misread this. 👍🏼

**Review by meganrogge — DISMISSED** [review]
Thank you 👍🏼

**Review by Tyriar — APPROVED** [review]

**Review by bpasero — APPROVED** [review]

**Pull request merged**